versioninfo.json.tmpl

```diff
@@ -0,0 +1,30 @@
+{{- $name := "chezmoi" -}}
+
+{{- $commitHash := output "git" "rev-parse" "HEAD" | trim -}}
+{{- if ne "" (output "git" "diff" "--stat" | trim) -}}
+{{-   $commitHash = printf "%s-dirty" $commitHash -}}
+{{- end -}}
+
+{{- $version := semver (output "git" "describe" "--abbrev=0" "--tags" | trim) -}}
+{{- $versionDict := dict "Major" $version.Major "Minor" $version.Minor "Patch" $version.Patch "Build" 0 -}}
+{{- $fFileVersion := dict "FileVersion" $versionDict -}}
+{{- $fProductVersion := dict "ProductVersion" $versionDict -}}
+{{- $fixedFileInfo := dict "FixedFileInfo" (merge $fFileVersion $fProductVersion) -}}
+
+{{- $filename := printf "%s.exe" $name -}}
+
+{{- /* Get the current year using the Magical Reference Date format */ -}}
+{{- $currentYear := now | date "2006" -}}
+
+{{- $sComments := dict "Comments" (printf "commit: %s" $commitHash) -}}
+{{- $sFileDescription := dict "FileDescription" "Manage your dotfiles across multiple diverse machines, securely." -}}
+{{- $sFileVersion := dict "FileVersion" $version.Original -}}
+{{- $sInternalName := dict "InternalName" $filename -}}
+{{- $sLegalCopyright := dict "LegalCopyright" (printf "Copyright (c) %s Tom Payne" $currentYear) -}}
+{{- $sOriginalFilename := dict "OriginalFilename" $filename -}}
+{{- $sProductName := dict "ProductName" $name -}}
+{{- $sProductVersion := dict "ProductVersion" $version.Original -}}
+{{- $stringFileInfo := dict "StringFileInfo" (merge $sComments $sFileDescription $sFileVersion $sInternalName $sLegalCopyright $sOriginalFilename $sProductName $sProductVersion) -}}
+
+{{- $outJson := merge $fixedFileInfo $stringFileInfo -}}
+{{- $outJson | toPrettyJson }}
```

Comment:
Please move this file to `assets/templates`. I'm trying to keep the top level directory as clean as possible because people have to scroll past it to read the README when they visit https://github.com/twpayne/chezmoi.

Reply:
I also think we can create the data structure more clearly by constructing the final dict directly and using fewer intermediate variables and dicts, something like:

```
{{- dict
      "FixedFileInfo" (dict
        "FileVersion" $versionDict
        "ProductVersion" $versionDict)
      "StringFileInfo" (dict
        "Comments" (printf "commit: %s" $commitHash))
    | toPrettyJson
-}}
```

Reply:
Done, also shuffled the variables around a bit to group similar ones.

I'm not sure why, but I didn't think it was possible to have a multi line action like that, otherwise I probably would have done that from the start.